contracts/generic/AugmintToken.sol

```diff
@@ -30,12 +33,39 @@ contract AugmintToken is AugmintTokenInterface {
         feeAccount = _feeAccount;
 
     }
-
     function transfer(address to, uint256 amount) external returns (bool) {
         _transfer(msg.sender, to, amount, "");
         return true;
     }
 
+    /* Transfers based on an offline signed transfer instruction. */
+    function delegatedTransfer(address from, address to, uint amount, string narrative,
+                                     uint minGasPrice,  /* client provided gasPrice on which she expects tx to be exec. */
+                                     uint maxExecutorFee, /* client provided max fee for executing the tx */
+                                     bytes32 nonce, /* random nonce generated by client */
+                                     /* ^^^^ end of signed data ^^^^ */
+                                     bytes signature,
+                                     uint requestedExecutorFee /* the executor can decide to request lower fee */
+                                     )
+    external returns(bool) {
+
+        require(!noncesUsed[nonce], "nonce already used");
+        require(tx.gasprice >= minGasPrice, "tx.gasprice must be >= minGasPrice");
```

Comment:
why do we care about the gas price at this point? the transaction is already being processed...

Reply:
as for now do you suggest to get rid of minGasprice ?

Reply:
Yes.